### PR TITLE
[Cleanup] Line Items - Auto Creation 

### PR DESCRIPTION
**Pull request by Civolilah**
@beganovich @turbo124 PR includes:

- Me and David were talking on Slack about a bug related to creating `Products and Tasks line items at the same time`. We had a bug in incorrectly updating field values ​​(the issue was indexing the items). So PR includes fixing it.
- Implementing auto-create line items related to behavior in a Flutter app.

I hope I covered all the cases where we had bugs and all the cases where we had to clean-up (create/delete) items related to the auto-clean part of the task.

Let me know your thoughts.

### CODE REVIEW

**Comment by turbo124** [comment]
@Civolilah 

A few issues here. On new invoice, two line items are inserted.
Also, when typing text into product and then tabbing, we lose the value that was inserted,

**Comment by Civolilah** [comment]
> @Civolilah
> 
> A few issues here. On new invoice, two line items are inserted. Also, when typing text into product and then tabbing, we lose the value that was inserted,

@turbo124 Both behaviors have been improved and fixed. I hope he looks satisfying from your side as well. Let me know your thoughts.

File: src/pages/invoices/common/hooks/useResolveInputField.tsx [hunk 1]
```diff
@@ -45,18 +49,82 @@ interface Props {
...
+        const lastLineItemIndex = lineItems.indexOf(
+          typeFilteredLineItems[lineItemsLength - 1]
+        );
+
+        if (lastLineItemIndex > -1) {
+          props.deleteLineItem(lastLineItemIndex);
+        }
+      }
+    }
+  };
```
Comment (beganovich) [inline src/pages/invoices/common/hooks/useResolveInputField.tsx:126]: This seems like a fairly expensive operation. Is it possible to memoize this with `useCallback`?

File: src/pages/purchase-orders/edit/hooks/useHandleLineItemPropertyChange.ts [hunk 2]
```diff
@@ -26,12 +26,20 @@ export function useHandleLineItemPropertyChange(
     value: unknown,
     index: number
   ) => {
+    const updatedPurchaseOrderLineItems = purchaseOrder.line_items;
+
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    updatedPurchaseOrderLineItems[index][property] = value;
+
```
Comment (beganovich) [inline src/pages/purchase-orders/edit/hooks/useHandleLineItemPropertyChange.ts:34]: Oof, can you explain why we do this?

**Review by beganovich — CHANGES_REQUESTED** [review]
Few (possible) changes are needed.

File: src/pages/invoices/common/hooks/useResolveInputField.tsx (hunk 1 above)
Comment (Civolilah) [reply src/pages/invoices/common/hooks/useResolveInputField.tsx]: Sure, the logic has been changed.

File: src/pages/purchase-orders/edit/hooks/useHandleLineItemPropertyChange.ts (hunk 2 above)
Comment (Civolilah) [reply src/pages/purchase-orders/edit/hooks/useHandleLineItemPropertyChange.ts]: The logic has been changed, so this is removed!

**Review by beganovich — APPROVED** [review]

**Pull request merged**